### DISCUSSION TITLE
Possibility to add additional data to contract's calldata

### DISCUSSION THREAD

**Discussion by happyeric77** [MetaMask/metamask-improvement-proposals#50] [General]
The project I am working on requires appending additional data to the smart contract `calldata` field.
However, it seems like Metamask puts a validation to check the calldata format.  I am not sure how it exactly works. 
I assume metamask looks for the corresponding abi for those validated contract. And use it to validate the calldata.
I am wondering if we can somehow get around it.
For more detail, here is [the discussion in StackOverflow](https://ethereum.stackexchange.com/questions/165171/any-way-that-i-can-append-a-data-along-with-a-evm-smart-contract-transaction)

**Comment by SyedAsadKazmi** [comment]
Yeah, I think it would be great if instead of just invalidating any extra data passed to the token transaction like `approve`, `transfer`, etc., the attached extra data should be shown to the user in the actual as well as decoded format, so as to retain the transparency of the transaction.
